Sources/MapboxMobileEvents/NSUserDefaults+MMEConfiguration.m

```diff
@@ -317,12 +318,38 @@ - (NSString *)mme_userAgentString {
              [NSProcessInfo mme_processorTypeDescription]
         ];
         
+        // try to get a list of loaded dylibs
+        struct task_dyld_info tdi;
+        mach_msg_type_number_t count = TASK_DYLD_INFO_COUNT;
```

Comment:
Why is this variable initialized to `TASK_DYLD_INFO_COUNT`. It seems like it's used to collect the count of the output from `task_info`, so could it just be initialized to 0?

Reply:
Also, is `count` always 1 after `task_info` returns?

Reply:
That is how the API works. Check `task_info.h` - there are many `*_INFO_COUNT` constants. To get a list of loaded dylibs, we have to pass this one.

Reply:
Also, there are some examples in `https://opensource.apple.com/source/xnu/xnu-4903.231.4/tests/task_info.c.auto.html`

Reply:
Interesting. Here's a link to the source for anyone who wants to read more… https://github.com/apple/darwin-xnu/blob/2ff845c2e033bd0ff64b5b6aa6063a1f8f65aa32/osfmk/kern/task.c

Seems like it's used to indicate the size/version of the 3rd argument. There's even an interesting workaround for a specific iOS version that used this API incorrectly… https://github.com/apple/darwin-xnu/blob/2ff845c2e033bd0ff64b5b6aa6063a1f8f65aa32/osfmk/kern/task.c#L5248-L5257